### ISSUE TITLE
[Feature Request]: Expose (group) channels to ATV Home Launcher.

### ISSUE THREAD

**Issue by subdancer** [yuliskov/SmartTube#3959]
### Describe the feature you'd like to request

At the moment smarttube exposes 3 channels to the Android TV Home Launcher.

Subscriptions -> updateOrPublishChannel(mService.getSubscriptionsPlaylist());
Home -> updateOrPublishChannel(mService.getRecommendedPlaylist());
History -> updateOrPublishChannel(mService.getHistoryPlaylist());

Being able to add the newly added group channels will make the Android TV Home Launcher/Homescreen way more customised and neat. 

### Describe the solution you'd like

Update the functions
updateChannels();
updateRecommendations();
in
https://github.com/yuliskov/SmartTube/blob/bc005d3f27ea70947130c9537ca20bbacdecf9d2/leanbackassistant/src/main/java/com/liskovsoft/leanbackassistant/channels/UpdateChannelsTask.java#L30-L65
to expose the newly introduced group channels to the android tv home launcher.

### Describe alternatives you've considered

Or maybe expose all channels that got pinned to smarttubes sidebar. Guess this will include group channels and also give the option that the user can have its favourite channels directly on the home screen when the device gets turned on.  I'm not sure if this is easy to implement or a big task.

**Comment by subdancer** [comment]
Yes we are able to select 3 channels (home,subscriptions and history) on the android tv launcher as feeds, that is all. 
Sorry for my bad english, but what i'm trying to communicate is that it would be great to have more than those 3 standard channels.  Basicly all channels that get added to smarttubes sidebar.

Here are some pictures to explain further...
In this one we can see that the latest codebase of smarttube exposes only 3 Channels to the ATV Launcher:
![Screenshot_20241002
Those are:
![Screenshot_20241002_222650](https://github.com/user-attachments/assets/9f279065-9912-422b-90a4-48d737f99cb0)
_222627](https://github.com/user-attachments/assets/68059292-b185-4e40-ab8c-88f037662c76)
![Screenshot_20241002_222650](https://github.com/user-attachments/assets/25e791a6-0bf3-4c70-98ea-d1946a1af83a)

I have added 2 channels to my smarttube sidebar on the left side as an example (see picture below).
Kartonok and another channel called Irony. The latter is a new feature that just got implemented called group channels. 
Basicly a channel that holds a selection channels (PS. being able to add a custom icon or symbol for the group channels would nice too but it's a feature that just got added in the latest release, so no rush).
![Screenshot_20241002_221405](https://github.com/user-attachments/assets/d783239f-5022-45d9-827f-186c56a1033a)

I want those 2 Channels (Kartonok and the group channel called Irony)  to show up in the channel selection of the android launcher. basicly over here:
![Screenshot_20241002_222650](https://github.com/user-attachments/assets/25e791a6-0bf3-4c70-98ea-d1946a1af83a)
That would give us the power to get rid of the standard channels if we want and make custom ones for the homescreen =)
For example a group channel called music, that holds only a selection of musicians or a channel called Cooking that holds ones favourite chef channels. 

In my opinion this would be an awesome feature, because we would be able to customize the android tv homescreen with channels that hold content we actually care about and we will see news about them right when we turn on the device.  I hope this clears up any missunderstandings. :bulb:

**Comment by tkhr123** [comment]
Hello, could you tell me which version of Android TV Home you're using? I noticed yours doesn't have any ads from Google, which I hate.

**Comment by subdancer** [comment]
> those are youtube API depedent not smartube so up to youtube to make more available to the tv recommendation service

@pucknbass I'm not talking about modifying yt's channels. This is not even about yt.. Simply another missunderstanding.

If one writes an app and agreggates a list of (video) content be it external or internal and adds it to a channel builder, one is able to provide that channel to the homescreen (to any launcher that supports channels). The user will then be able to select and add that channel to the launcher.  See here: https://developer.android.com/training/tv/discovery/recommendations-channel#app_channels
In my oponion auto picking those that get added to the st's sidebar would be a neat solution, because that way only those that the user actually cares about will be available.

@tkhr123 sure 
![atv](https://github.com/user-attachments/assets/219bb823-549a-4ab8-8ece-3a4afa6b6b2a)
